lib/aws-record/record/batch.rb

```diff
@@ -0,0 +1,73 @@
+# Copyright 2015-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License is
+# located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+
+module Aws
+  module Record
+    module Batch
+      def self.included(base)
```

Comment:
Overall I really like this approach as it decouples some of the error/mess nicely.  

My one general comment is that I think the batch write functionality should maybe be independent of individual models - the code for the `BatchWrite` you have already accounts for this (by tracking the changes per table).  

Instead of including the Batch module in the model class, you just instantiate one (and provide a ddb client, or rely on the same creation logic the aws record module uses) and then do the same put/delete actions on it.  This ends up being a bit more natural for operations against multiple tables and including the module doesn't actually give us anything aside from access to the client.

Reply:
Sure! That's a good suggestion. Do you prefer the class instantiated to be named `Aws::Record::Batch`?